dataservice_publisher/app.py

```diff
@@ -44,6 +44,35 @@ async def authenticated(request: web.Request) -> bool:
     return False
 
 
+@web.middleware
+async def content_negotiation_middleware(
+    request: web.Request, handler: Any
+) -> web.Response:
+    """Middleware to check if we can respond in accordance to the user's accept-header."""
+    logging.debug("content_negotiation_middleware called")
+
+    if not request.headers.getone(hdrs.ACCEPT, None):
+        pass  # pragma: no cover
+    elif "catalogs" in request.path:
+        content_types = request.headers.getone(hdrs.ACCEPT).split(",")
```

Comment:
I eksempelet her er `, ` (dot  + space) delimiter: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept

Reply:
Bør kanskje sjekke alle headers og ikke bar _first_? Man kan finne på å skrive `-H "Accept: foo/bar" -H "Accept: text/turtle"`

Reply:
Re space: sidan eg sjekkar med `"foo" _in_ bar`, ville det ha funka, men eg har fjerna alle spaces, just in case.
Re mange headers, har fiksa i 19df9d7

Reply:
Re Re space: om du sjekker `"foo" in headers.split(",")` og `headers` er `"bar, foo"` vil det bli `"foo" in ["bar", " foo"]`, som er `False` fordi det er en ekstra space i `" foo"`.

Reply:
Refaktorert i ae0a46e0a5005369fd6a0269189b0e02d1e937e1 Trur kommentaren er irrelevant no, men fint om du tar ein sjekk.